src/main/java/com/teamc/fms/controller/TimeSlotController.java

```diff
@@ -0,0 +1,62 @@
+package com.teamc.fms.controller;
+
+
+import com.teamc.fms.dto.TimeSlotDto;
+import com.teamc.fms.entity.TimeSlot;
+import com.teamc.fms.mapper.TimeSlotMapper;
+import com.teamc.fms.service.TimeSlotService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/v1")
+public class TimeSlotController {
+    private final TimeSlotService timeSlotService;
+    private final TimeSlotMapper timeSlotMapper;
+
+
+    @PostMapping("/time-slots")
+    @ResponseStatus(HttpStatus.CREATED)
+    public TimeSlotDto addTimeSlot(@RequestBody TimeSlotDto timeSlot) {
+        return timeSlotService.addTimeSlot(timeSlot);
+    }
+
+    @GetMapping("/time-slots")
+    @ResponseStatus(HttpStatus.OK)
+    public List<TimeSlotDto> getAllTimeSlots() {
+        List<TimeSlot> timeSlots = timeSlotService.getAllTimeSlots();
+        return timeSlots.stream().map(timeSlotMapper::toTimeSlotDto).toList();
+    }
+
+    @GetMapping("/time-slots/{id}")
+    public ResponseEntity<Optional<TimeSlot>> getUserById(@PathVariable("id") String id) {
```

Comment:
Do not return Optional
in case there is no timeslot with this ID, you should return 404